gms/gossiper.cc

```diff
@@ -632,6 +633,11 @@ future<> gossiper::apply_state_locally_without_listener_notification(std::unorde
 }
 
 future<> gossiper::apply_state_locally(std::map<inet_address, endpoint_state> map) {
+    if (_apply_state_locally_semaphore.waiters() > gossiper::apply_max_queue_size && !should_count_as_msg_processing(map)) {
```

Comment:
We already call should_count_as_msg_processing everywhere before we call apply_state_locally. Perhaps we can pass a flag to avoid calling should_count_as_msg_processing again.

Reply:
Ok, will propagate it. Indeed, it is not a trivial check, so worth trying to save the effort.